python/ray/air/checkpoint.py

```diff
@@ -234,20 +227,24 @@ def from_dict(cls, data: dict) -> "Checkpoint":
         Returns:
             Checkpoint: checkpoint object.
         """
+        if isinstance(data, CheckpointDict):
+            cls = data.checkpoint_cls
```

Comment:
We should document this in the docstring of from_dict - otherwise users could be confused that they don't get a pure `Checkpoint` object back.

Also, we should make sure that this only triggers in the base checkpoint class - if I do TorchCheckpoint.from_dict() I would want to get a TorchCheckpoint back in any case (e.g. if I pass a regular Checkpoint-CheckpointDict).